### PR TITLE
sharness: add t0040-install-many.sh

### PR DESCRIPTION
**Pull request by chriscool**
This is to test installing different ipfs versions.

If the added tests pass we can close issue #7 (Timeouts with v0.3.8 and v0.3.7).

### CODE REVIEW

**Comment by jbenet** [comment]
seems like it failed. cc @whyrusleeping

**Comment by chriscool** [comment]
Yeah, it fails with:

```
expecting success: 

		exec_docker "$DOCID" "$GUEST_IPFS_UPDATE --verbose install $VERSION" >actual 2>&1 ||

		test_fsh cat actual

	

> cat actual

daemon check failed: open /root/.ipfs/api: no such file or directory

installing ipfs version v0.3.10

  - using GOOS=linux and GOARCH=amd64

  - fetching "/ipns/update.ipfs.io/go-ipfs/v0.3.10/go-ipfs_v0.3.10_linux-amd64.zip"

fetching url: https://ipfs.io/ipns/update.ipfs.io/go-ipfs/v0.3.10/go-ipfs_v0.3.10_linux-amd64.zip

  - writing to /tmp/ipfs-update030874052/go-ipfs_v0.3.10_linux-amd64.zip

  - extracting binary to tempdir:  /tmp/ipfs-update480963513/ipfs-new

binary downloaded, verifying...

  - running init in '/root/.ipfs/update-staging/test196113491' with new binary

  - checking new binary outputs correct version

  - tweaking test config to avoid external interference

  - starting up daemon

  - waiting on daemon to come online

  - found api file

  - checking that we can add and cat a file

  - checking that file shows up in ipfs refs local

  - killing test daemon

success!

stashing old binary

  - moving /usr/local/bin/ipfs to /root/.ipfs/old-bin/ipfs-v0.3.7

installing new binary to /usr/local/bin/ipfs

checking if repo migration is needed...

ERROR: Migration Failed:  open /root/.ipfs/version: no such file or directory

install failed, reverting changes...

ERROR: open /root/.ipfs/version: no such file or directory 

not ok 6 - 'ipfs-update install' works for v0.3.10
```

It also fails locally on my machine.

**Comment by chriscool** [comment]
I just added revert tests that also fail.

**Comment by chriscool** [comment]
@whyrusleeping it looks to me that the following code in migrations.go should be improved:

```
func CheckMigration() error {
	stump.Log("checking if repo migration is needed...")
	p := util.IpfsDir()
	oldverB, err := ioutil.ReadFile(filepath.Join(p, "version"))
	if err != nil {
		return err
	}
...
```

When v0.3.7 is installed the version file is not created. It is only created when "ipfs init" is run.

**Comment by chriscool** [comment]
I fixed a few things in migrations.go main.go and revert.go, so that the tests now pass.

File: migrations.go [hunk 1]
```diff
@@ -14,13 +12,11 @@ import (
 
 func CheckMigration() error {
 	stump.Log("checking if repo migration is needed...")
-	p := util.IpfsDir()
-	oldverB, err := ioutil.ReadFile(filepath.Join(p, "version"))
+	oldver, err := GetCurrentVersion()
```
Comment (whyrusleeping) [inline migrations.go]: I don't know if this will work right. The version returned here is the ipfs binary version, and we're looking for the version of the repo.

**Comment by whyrusleeping** [comment]
@chriscool i'll push a fix to this branch for you for the repo version thing.

File: migrations.go (hunk 1 above)
Comment (chriscool) [inline migrations.go]: Erroring out when the "version" file is not there was wrong anyway, because the "version" file is created by some versions of IPFS only when `ipfs init` is run.

Comment (whyrusleeping) [inline migrations.go]: agreed. we should have some function that maps ipfs version to a repo version for the versions we can't call `ipfs version --repo` on.

Comment (chriscool) [inline migrations.go]: Yeah, but maybe that is a separate topic and you can have a different PR do that.

Comment (whyrusleeping) [inline migrations.go]: actually, now that i think about it. if theres no version file, theres no repo to migrate. So we can just short circuit out of this check in that case.

Comment (whyrusleeping) [inline migrations.go]: can change it to something like:
```
vfilePath := filepath.Join(p, "version")
if _, err := os.Stat(vfilePath); err == os.ErrNotExist {
    VLog("  - no prexisting repo to migrate")
    return nil
}
```

Comment (chriscool) [inline migrations.go]: Ok, I just pushed a new version of this patch series that does something like that.

**Comment by chriscool** [comment]
I had to push another version of the series, because as PR #15 had been merged, tests didn't pass anymore.

**Comment by whyrusleeping** [comment]
this LGTM
